### PR TITLE
Implement GET_ACTION_AVAIL and GET_NOTIF_SIZES for seccomp

### LINKED ISSUE #2903: Assertion from seccomp
I was trying to debug a crash in [the mozregression-gui release binary](https://github.com/mozilla/mozregression/releases/download/4.0.18/mozregression-gui.tar.gz) and got a crash in rr... Seems that there's some `seccomp` arguments that isn't supported?

Full output.

```
[FATAL /usr/src/debug/rr/src/record_syscall.cc:5929:rec_process_syscall_arch()] 
 (task 3816514 (rec:3816514) at time 4352)
 -> Assertion `t->regs().syscall_result_signed() == -syscall_state.expect_errno' failed to hold. Expected EINVAL for 'seccomp' but got result 0 (errno SUCCESS); unknown seccomp(0x2)
Tail of trace dump:
{
  real_time:357681.154777 global_time:4332, event:`SYSCALL: rrcall_init_buffers' (state:EXITING_SYSCALL) tid:3816514, ticks:91511
rax:0x7fd5e1158000 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x681ffe70 rbp:0x3 rsp:0x681ffe20 r8:0x0 r9:0x0 r10:0x0 r11:0x246 r12:0x64 r13:0x681ffe70 r14:0x3a3c42 r15:0x0 rip:0x70000005 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x3e9 fs_base:0x7fd5e1258c00 gs_base:0x0
  { map_file:"<ZERO>", addr:0x7fd5e1158000, length:0x100000, prot_flags:"rw-s", file_offset:0x0, device:40, inode:46107, data_file:"", data_offset:0x0, file_size:0x100000 }
  { tid:3816514, addr:0x7fd5e1b4022c, length:0x1 }
  { tid:3816514, addr:0x7fd5e1b40302, length:0x1 }
  { tid:3816514, addr:0x681ffe70, length:0x20 }
}
{
  real_time:357681.154811 global_time:4333, event:`SYSCALLBUF_FLUSH' tid:3816514, ticks:91745
  { syscall:'futex', ret:0x0, size:0x14 }
}
{
  real_time:357681.154820 global_time:4334, event:`SYSCALL: brk' (state:ENTERING_SYSCALL) tid:3816514, ticks:91745
rax:0xffffffffffffffda rbx:0x21000 rcx:0xffffffffffffffff rdx:0x2b0 rsi:0x7fd5e1a76a00 rdi:0x0 rbp:0x7fd5e1a7a420 rsp:0x7ffe8eff9a08 r8:0x3 r9:0x7fd5e1a76a60 r10:0xfffffffffffff000 r11:0x246 r12:0x0 r13:0x21000 r14:0x7fd5e1a76a60 r15:0x2b0 rip:0x7fd5e19ab4ab eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xc fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.154823 global_time:4335, event:`SYSCALLBUF_RESET' tid:3816514, ticks:91745
}
{
  real_time:357681.154843 global_time:4336, event:`SYSCALL: brk' (state:EXITING_SYSCALL) tid:3816514, ticks:91745
rax:0x5603b6995000 rbx:0x21000 rcx:0xffffffffffffffff rdx:0x2b0 rsi:0x7fd5e1a76a00 rdi:0x0 rbp:0x7fd5e1a7a420 rsp:0x7ffe8eff9a08 r8:0x3 r9:0x7fd5e1a76a60 r10:0xfffffffffffff000 r11:0x246 r12:0x0 r13:0x21000 r14:0x7fd5e1a76a60 r15:0x2b0 rip:0x7fd5e19ab4ab eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xc fs_base:0x7fd5e1258c00 gs_base:0x0
  { map_file:"<ZERO>", addr:0x5603b6995000, length:(nil), prot_flags:"---p", file_offset:0x0, device:0, inode:0, data_file:"", data_offset:0x0, file_size:0x0 }
}
{
  real_time:357681.154869 global_time:4337, event:`SYSCALL: brk' (state:ENTERING_SYSCALL) tid:3816514, ticks:91750
rax:0xffffffffffffffda rbx:0x5603b69b6000 rcx:0xffffffffffffffff rdx:0x7fd5e1a7a420 rsi:0x7fd5e1a76a00 rdi:0x5603b69b6000 rbp:0x7fd5e1a7a420 rsp:0x7ffe8eff9a08 r8:0x3 r9:0x7fd5e1a76a60 r10:0xfffffffffffff000 r11:0x246 r12:0x5603b6995000 r13:0x21000 r14:0x7fd5e1a76a60 r15:0x2b0 rip:0x7fd5e19ab4ab eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xc fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.154925 global_time:4338, event:`SYSCALL: brk' (state:EXITING_SYSCALL) tid:3816514, ticks:91750
rax:0x5603b69b6000 rbx:0x5603b69b6000 rcx:0xffffffffffffffff rdx:0x7fd5e1a7a420 rsi:0x7fd5e1a76a00 rdi:0x5603b69b6000 rbp:0x7fd5e1a7a420 rsp:0x7ffe8eff9a08 r8:0x3 r9:0x7fd5e1a76a60 r10:0xfffffffffffff000 r11:0x246 r12:0x5603b6995000 r13:0x21000 r14:0x7fd5e1a76a60 r15:0x2b0 rip:0x7fd5e19ab4ab eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xc fs_base:0x7fd5e1258c00 gs_base:0x0
  { map_file:"<ZERO>", addr:0x5603b6995000, length:0x21000, prot_flags:"rw-p", file_offset:0x0, device:0, inode:0, data_file:"", data_offset:0x0, file_size:0x21000 }
}
{
  real_time:357681.154999 global_time:4339, event:`PATCH_SYSCALL' tid:3816514, ticks:93306
rax:0x9d rbx:0x7ffe8eff9ec8 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x26 rbp:0x5603b524d8f8 rsp:0x7ffe8eff9c90 r8:0x0 r9:0x0 r10:0x0 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x7fd5e19b4e1f eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xffffffffffffffff fs_base:0x7fd5e1258c00 gs_base:0x0
  { tid:3816514, addr:0x7fd5e1b7c0b5, length:0x4f }
  { tid:3816514, addr:0x7fd5e19b4e1f, length:0x5 }
  { tid:3816514, addr:0x7fd5e19b4e24, length:0x3 }
}
{
  real_time:357681.155027 global_time:4340, event:`SYSCALL: prctl' (state:ENTERING_SYSCALL) tid:3816514, ticks:93314
rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x26 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x0 r9:0x0 r10:0x0 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x9d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155045 global_time:4341, event:`SYSCALL: prctl' (state:EXITING_SYSCALL) tid:3816514, ticks:93314
rax:0x0 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x26 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x0 r9:0x0 r10:0x0 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x9d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155069 global_time:4342, event:`SYSCALL: prctl' (state:ENTERING_SYSCALL) tid:3816514, ticks:93328
rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x4 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x0 r9:0x0 r10:0x0 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x9d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155086 global_time:4343, event:`SYSCALL: prctl' (state:EXITING_SYSCALL) tid:3816514, ticks:93328
rax:0x0 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x0 rdi:0x4 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x0 r9:0x0 r10:0x0 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x9d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155143 global_time:4344, event:`PATCH_SYSCALL' tid:3816514, ticks:93347
rax:0x13d rbx:0x13d rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x0 rbp:0x0 rsp:0x7ffe8eff9cb8 r8:0xa r9:0x7ffe8eff9ec8 r10:0xc9 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x7fd5e19af18b eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0xffffffffffffffff fs_base:0x7fd5e1258c00 gs_base:0x0
  { tid:3816514, addr:0x7fd5e1b7c104, length:0x4f }
  { tid:3816514, addr:0x7fd5e19af18b, length:0x5 }
  { tid:3816514, addr:0x7fd5e19af190, length:0x3 }
}
{
  real_time:357681.155170 global_time:4345, event:`SYSCALL: seccomp' (state:ENTERING_SYSCALL) tid:3816514, ticks:93355
rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x0 rbp:0x681fffa0 rsp:0x681ffd60 r8:0xa r9:0x7ffe8eff9ec8 r10:0xc9 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155188 global_time:4346, event:`SYSCALL: seccomp' (state:EXITING_SYSCALL) tid:3816514, ticks:93355
rax:0xffffffffffffffea rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x0 rbp:0x681fffa0 rsp:0x681ffd60 r8:0xa r9:0x7ffe8eff9ec8 r10:0xc9 r11:0x246 r12:0x3 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155211 global_time:4347, event:`SYSCALL: seccomp' (state:ENTERING_SYSCALL) tid:3816514, ticks:93374
rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x1 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x7ffe8eff9ec8 r9:0x3 r10:0xa r11:0x246 r12:0x1 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155250 global_time:4348, event:`SYSCALL: seccomp' (state:EXITING_SYSCALL) tid:3816514, ticks:93374
rax:0xfffffffffffffff2 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x1 rdi:0x1 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x7ffe8eff9ec8 r9:0x3 r10:0xa r11:0x246 r12:0x1 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155276 global_time:4349, event:`SYSCALL: seccomp' (state:ENTERING_SYSCALL) tid:3816514, ticks:93392
rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x2 rdi:0x1 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x3 r9:0x3 r10:0x7ffe8eff9ec8 r11:0x246 r12:0x2 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155314 global_time:4350, event:`SYSCALL: seccomp' (state:EXITING_SYSCALL) tid:3816514, ticks:93392
rax:0xfffffffffffffff2 rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x0 rsi:0x2 rdi:0x1 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x3 r9:0x3 r10:0x7ffe8eff9ec8 r11:0x246 r12:0x2 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
{
  real_time:357681.155339 global_time:4351, event:`SYSCALL: seccomp' (state:ENTERING_SYSCALL) tid:3816514, ticks:93415
rax:0xffffffffffffffda rbx:0x681fffa0 rcx:0xffffffffffffffff rdx:0x7ffe8eff9cac rsi:0x0 rdi:0x2 rbp:0x681fffa0 rsp:0x681ffd60 r8:0x3 r9:0x7ffe8eff9cb0 r10:0x3 r11:0x246 r12:0x1 r13:0x0 r14:0x7ffe8eff9d84 r15:0x5603b524f8a0 rip:0x70000002 eflags:0x246 cs:0x33 ss:0x2b ds:0x0 es:0x0 fs:0x0 gs:0x0 orig_rax:0x13d fs_base:0x7fd5e1258c00 gs_base:0x0
}
=== Start rr backtrace:
rr(_ZN2rr9GdbServer15emergency_debugEPNS_4TaskE+0x97d)[0x564def5e6aad]
rr(+0xb536e)[0x564def5f536e]
rr(+0xb55ed)[0x564def5f55ed]
rr(+0x1153af)[0x564def6553af]
rr(_ZN2rr19rec_process_syscallEPNS_10RecordTaskE+0x31d)[0x564def63082d]
rr(_ZN2rr13RecordSession21syscall_state_changedEPNS_10RecordTaskEPNS0_9StepStateE+0x3e2)[0x564def618792]
rr(_ZN2rr13RecordSession11record_stepEv+0x479)[0x564def620f99]
rr(_ZN2rr13RecordCommand3runERSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EE+0xb36)[0x564def612cf6]
rr(main+0x1a1)[0x564def588541]
/usr/lib/libc.so.6(__libc_start_main+0xd5)[0x7fcf33422b25]
rr(_start+0x2e)[0x564def58870e]
=== End rr backtrace
```

### PR DESCRIPTION
**Pull request by yuyichao**
Fixes #2903

### CODE REVIEW

**Comment by rocallahan** [comment]
Let's not merge this until the remaining crash in replay in https://github.com/rr-debugger/rr/issues/2903 has been diagnosed.

File: src/test/seccomp.c [hunk 1]
```diff
@@ -131,12 +131,38 @@ static void* run_thread(__attribute__((unused)) void* p) {
...
+    uint16_t seccomp_notif_resp;
+    uint16_t seccomp_data;
+  } sizes;
+  int ret = syscall(RR_seccomp, SECCOMP_GET_NOTIF_SIZES, 0, &sizes);
+  assert(ret == EINVAL || ret == 0);
+  if (ret == 0) {
+    // These were the sizes when `SECCOMP_GET_NOTIF_SIZES` was first added.
+    assert(sizes.seccomp_notif >= 80);
+    assert(sizes.seccomp_notif_resp >= 24);
+    assert(sizes.seccomp_data >= 64);
```
Comment (rocallahan) [inline src/test/seccomp.c:153]: These should be test_assert.

**Review by rocallahan — CHANGES_REQUESTED** [review]

File: src/test/seccomp.c (hunk 1 above)
Comment (yuyichao) [reply src/test/seccomp.c:153]: Huh, don't know why I missed that. I assume all the other added ones should be as well.
I did check that a failure here trigger a test failure at least...

Comment (yuyichao) [reply src/test/seccomp.c]: Done.

**Pull request merged**